File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>>nodes;
        
        if(root ==NULL) return nodes;
        
        queue<TreeNode*>q;
        q.push(root);
        
        while(!q.empty()){
            int currentSize = q.size();
            vector<int>levelNodes;
            while(currentSize > 0){
                TreeNode* currentNode = q.front();
                q.pop();
                
                //Add to levelNode
                levelNodes.push_back(currentNode->val);
                
                if(currentNode->left != NULL)
                    q.push(currentNode->left);
                if(currentNode->right != NULL)
                    q.push(currentNode->right);
                
                currentSize--;
            }
            nodes.push_back(levelNodes);
        }
        
        return nodes;
    }
};

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root)
    {
        
        List<List<Integer>> answer = new ArrayList<>();
    
        Queue<TreeNode> queue = new LinkedList<>();
        
        if(root == null)
            return answer;
        
        queue.add(root);
    
        while(!queue.isEmpty())
        {
            
            int currentSize = queue.size();
            List<Integer> currentAns = new ArrayList<>();
            
            while(currentSize > 0)
            {
                TreeNode currentNode = queue.remove();
                currentAns.add(currentNode.val);
            
                if(currentNode.left != null)
                    queue.add(currentNode.left);
            
                if(currentNode.right != null){
                    queue.add(currentNode.right);
            }
                currentSize--;
        }
            
        answer.add(currentAns);
            
    }
        
    return answer;
        
    }
}